src/components/TabGroup/index.tsx

```diff
@@ -0,0 +1,181 @@
+      if (isValidElement<{ name?: string }>(child) && child.props.name) {
+        return child.props.name === selected.toString()
+      }
+
+      return index === Number(selected)
+    },
+  )
+
+const computeBarProperties = (tabsWidth: number[], index: number) => {
+  if (!tabsWidth.length) return [0, 0]
+  const width = tabsWidth[index]
+  const left = tabsWidth.reduce((acc, cur, idx) => {
+    if (idx < index) {
+      return acc + cur
+    }
+
+    return acc
+  }, 0)
+
+  return [width, left]
+}
+
+type TabGroupProps = HTMLAttributes<HTMLElement> & {
+  selected?: number | string
+  onChange?: (data: unknown) => void
+}
+
+const TabGroup: FunctionComponent<TabGroupProps> & { Tab: typeof Tab } = ({
+  children = null,
+  selected,
+  onChange = () => {},
+  className = '',
+  ...props
+}) => {
+  const flattenedChildren = flattenChildren(children)
+  const [tabsWidth, setTabsWidth] = useState([])
+  const setInternTabsWidth = useCallback(
+    (width: number, index: number) =>
+      setTabsWidth((prev: number[]) => {
+        const all: number[] = [...prev]
+        all[index] = width
+
+        return all as never[]
+      }),
+    [setTabsWidth],
+  )
+
+  const isTabsWidthSet = tabsWidth.length === flattenedChildren.length
+
+  const currentTabIndex =
+    selected !== undefined && React.Children.count(children) > 0
+      ? getCurrentTabIndex(selected, flattenedChildren)
+      : -1
+
+  const [width, left] = computeBarProperties(tabsWidth, currentTabIndex)
+
+  const navigateWithArrow: KeyboardEventHandler<HTMLInputElement> = ({
+    currentTarget,
+    code,
+  }) => {
+    const activeTab = '[role="tab"]:focus'
+    const enabledTab = '[role="tab"]:not([aria-disabled="true"]'
+    if (code === 'ArrowLeft') {
+      const previousTab =
+        (currentTarget.querySelector(activeTab)
+          ?.previousElementSibling as HTMLElement) ||
+        [...Array.from(currentTarget.querySelectorAll(enabledTab))].pop()
+      previousTab?.focus()
+    } else if (code === 'ArrowRight') {
+      const nextTab =
+        (currentTarget.querySelector(
+          `${activeTab} ~ ${enabledTab}`,
+        ) as HTMLElement) ||
+        (currentTarget.querySelector(enabledTab) as HTMLElement)
+      nextTab?.focus()
+    }
+  }
+
+  return (
+    <Box {...props} position="relative">
+      <StyledTabs
+        role="tablist"
+        className={className}
+        onKeyDown={navigateWithArrow}
+      >
+        {flattenedChildren.map((child, index) => {
+          if (isValidElement<TabProps>(child)) {
+            const isSelected = child.props.name
+              ? child.props.name === selected
+              : index === selected
+
+            return cloneElement(child, {
+              index,
+              isSelected,
+              isTabsWidthSet,
+              key: child.props.name || index,
+              onClick: (
+                event:
+                  | MouseEvent<HTMLSpanElement>
+                  | KeyboardEvent<HTMLSpanElement>,
```

Comment:
```suggestion
```

Reply:
Need it because when he press Space or Enter we forward the event to the onClick handler, maybe better to separate onClick and onKeyDown handlers ? @adriengibrat

Reply:
I did something better IMO